tests/org.eclipse.reddeer.ui.test/src/org/eclipse/reddeer/ui/test/wizard/RedDeerTestCaseWizardInterfaceTest.java

```diff
@@ -0,0 +1,99 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat, Inc - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.reddeer.ui.test.wizard;
+
+import static org.junit.Assert.assertTrue;
+import java.util.List;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.JavaProjectWizard;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewInterfaceCreationWizard;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewInterfaceWizardPage;
+import org.eclipse.reddeer.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageOne;
+import org.eclipse.reddeer.eclipse.selectionwizard.NewMenuWizard;
+import org.eclipse.reddeer.eclipse.ui.problems.Problem;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.eclipse.reddeer.ui.test.wizard.impl.RedDeerTestCaseWizard;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class RedDeerTestCaseWizardInterfaceTest extends RedDeerWizardTestCase {
+
+	protected static final String PROJECT_NAME = "testproject";
+	protected static final String INTERFACE_NAME = "InterfaceName";
+	protected static final String SOURCE_FOLDER_NAME = PROJECT_NAME + "/src";
+	protected static final String PACKAGE = "org.example.test";
+	protected static NewMenuWizard wizard;
+
+	@Override
+	public String getWizardText() {
+		return "New RedDeer Test Case";
+	}
+	
+	@BeforeClass
+	public static void setup() {
+		wizard = new RedDeerTestCaseWizard();
+		projectName = PROJECT_NAME;
+		createTestProject();
+	}
+	
+	@Test
+	public void testInterfaceCreatedSuccessfully() {
+		// open and fill the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceWizardPage interfacePage = new NewInterfaceWizardPage(interfaceDialog);
+		interfacePage.setName(INTERFACE_NAME);
+		interfacePage.setPackage(PACKAGE);
+		interfacePage.setSourceFolder(SOURCE_FOLDER_NAME);
+
+		// test values in fields
+		assertTrue("Interface name optained from the input using getName() function is not the same as name set using setName().", interfacePage.getName().equals(INTERFACE_NAME));
+		assertTrue("Package name optained from the input using getPackage() function is not the same as name set using setPackage().", interfacePage.getPackage().equals(PACKAGE));
+		assertTrue("Source folder name optained from the input using getSourceFolder() function is not the same as name set using setSourceFolder().", interfacePage.getSourceFolder().equals(SOURCE_FOLDER_NAME));
+
+		// finish the wizard
+		interfaceDialog.finish();
+		
+		// test the Problem view for errors
+		ProblemsView problemsView = new ProblemsView();
```

Comment:
Why is good to check problems view when one has successfully finished wizard dialog which already got some validation? I would imagine testing the when dialog.finish was called, new Java editor was opened, that the editor contains proper package location and maybe keyword interface? Maybe even checking that chosen extending interfaces are there as well? Maybe more test cases?

Reply:
You are right, I have removed the problem view testing. Fixed.